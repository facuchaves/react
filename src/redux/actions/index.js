export const ACTIONS = {
  SEARCH_ISSUES: "SEARCH_ISSUES",
};

export const searchIssues = ( query ) => {
  return {
    type: ACTIONS.SEARCH_ISSUES,
    payload: query,
  };
};