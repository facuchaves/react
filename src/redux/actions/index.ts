export const ACTIONS = {
  SEARCH_ENTITIES: 'SEARCH_ENTITIES',
  SEARCH_ENTITY: 'SEARCH_ENTITY',
  CREATE_ENTITY: 'CREATE_ENTITY',
  UPDATE_ENTITY: 'UPDATE_ENTITY',
  DELETE_ENTITY: 'DELETE_ENTITY',
};

export const searchEntities = (query: any) => ({
  type: ACTIONS.SEARCH_ENTITIES,
  payload: {
    query,
  },
});

export const searchEntity = (entityId: number) => ({
  type: ACTIONS.SEARCH_ENTITY,
  payload: {
    entityId,
  },
});

export const createEntity = (entity: any) => ({
  type: ACTIONS.CREATE_ENTITY,
  payload: {
    entity,
  },
});

export const updateEntity = (entity: any) => ({
  type: ACTIONS.UPDATE_ENTITY,
  payload: {
    entity,
  },
});

export const deleteEntity = (entityId: number) => ({
  type: ACTIONS.DELETE_ENTITY,
  payload: {
    entityId,
  },
});
