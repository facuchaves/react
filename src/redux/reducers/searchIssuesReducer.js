import { ACTIONS } from "../actions";

const INITIAL_STATE = {
  query: {
    repo:"facebook/react",
    q:"",
    state:"OPEN"
  }
};

const searchIssuesReducer = (state = INITIAL_STATE, action) => {
  console.log("action.payload : " , action.payload)
  console.log("state : " , state)
  switch (action.type) {
    case ACTIONS.SEARCH_ISSUES:
      return {
        query: {
          repo:state.query.repo,
          q:action.payload.q,
          state:action.payload.state,
        }
      };
    default:
      return state;
  }
};

export default searchIssuesReducer;
