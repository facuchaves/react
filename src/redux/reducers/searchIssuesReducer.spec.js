import { ACTIONS } from "../actions";
import searchIssuesReducer from "./searchIssuesReducer";

test('searchIssuesReducer', () => {
  const state = {q:'Query',status:"OPEN"}
  expect( searchIssuesReducer( state, { type:ACTIONS.SEARCH_ISSUES , payload: state} ) ).toMatchSnapshot();
})