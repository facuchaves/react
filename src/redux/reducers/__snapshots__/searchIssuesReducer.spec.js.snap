// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchIssuesReducer 1`] = `
Object {
  "q": "Query",
  "query": Object {
    "q": "Query",
    "status": "OPEN",
  },
  "status": "OPEN",
}
`;
